### PR TITLE
Add Insomnia config file

### PR DESCRIPTION
**Pull request by smcclure17**
Adds an Insomnia config file, minor README updates.

### CODE REVIEW

File: functions/src/error-handling.ts [hunk 1]
```diff
@@ -30,17 +30,10 @@ export enum ShareLinkErrorCode {
   UNEXPECTED_ERROR,
 }
 
-interface ShareLinkErrorContent {
```
Comment (smcclure17) [inline functions/src/error-handling.ts:33]: Drive-by removing this type because it doesn't seem useful

File: README.md [hunk 2]
```diff
@@ -67,7 +70,7 @@ Returns new or existing share link for the URL provided with meta tags according
 ```
 
 ```bash
-curl -X POST -H "Content-Type:application/json" <baseurl>/api/registerUrl -d @./payload.json
+curl -v POST -H "Content-Type:application/json" https://us-central1-act-now-links-dev.cloudfunctions.net/api/registerUrl -d @./payload.json
```
Comment (mikelehen) [inline README.md]: I think this should b `-v -X POST` or similar.  As-is it logs:

> Warning: POST.
> * Could not resolve host: POST
> * Closing connection 0

at the top...

File: README.md [hunk 3]
```diff
@@ -84,21 +87,21 @@ object if none exist.
...
-* **Code:** `200 <br />`
-* **Content:** `{ urls: {share-link-url: {share-link-fields}} }`
+* **Code:** `200`
+* **Content:** `{ urls: {share-link-url: { share-link-fields }} }`
 
 #### Example
 
 ```bash
-curl -X GET "https://us-central1-act-now-links-dev.cloudfunctions.net/api/getShareLinkUrl?url=https://www.covidactnow.org"
+curl -v GET "https://us-central1-act-now-links-dev.cloudfunctions.net/api/shareLinksByUrl?url=https://www.covidactnow.org"
```
Comment (mikelehen) [inline README.md]: same here

File: README.md [hunk 4]
```diff
@@ -161,6 +164,13 @@ This project uses Firestore and Functions, so we will only need to configure emu
...
+Deploys are handled automatically by the [Firebase functions deploy](https://github.com/covid-projections/act-now-links-service/actions/workflows/functions-deploy.yml) Github action. The workflow is triggered on pushes to the `main` and `develop` branches so that the deployed functions will reflect the most up-to-date status of the repository.
 
 If need be, you can deploy functions yourself by running `yarn deploy` in `functions/`, but this is generally discouraged as it disrupts the symmetry between Github and Firebase.
+
+
+## Insomnia
+
+Insomnia can be used to test, debug, and store requests. To download and install the Insomnia client see https://insomnia.rest/download. 
+
+To work with the Act Now Links Service API in Insomnia, from the Insomnia dashboard select `Create > Import From > File` and select [`functions/insomnia-config.yaml`](./functions/insomnia-config.yaml).
```
Comment (mikelehen) [inline README.md]: Maybe I'm confused, but I don't see "Create" anywhere in the UI:
image

There's no "Create" menu either. I have Insomnia, Edit, View, Window, Tools, Help.

Comment (mikelehen) [reply README.md]: Maybe I installed / used the wrong thing?

**Review by mikelehen — CHANGES_REQUESTED** [review]
I didn't look at the insomnia config, but a couple drive-by comments on the docs.

File: README.md (hunk 4 above)
Comment (smcclure17) [reply README.md]: Ah, thanks for testing--that's the right client and you're right!

I was looking at this screen (below), but the screen you posted is the home/default screen. 
image

There's a way to import data through the Insomnia menu, so I've updated this to document that

File: README.md (hunk 2 above)
Comment (smcclure17) [reply README.md]: Ah, oops

Thanks, using `-v -X POST ...`  logs `Note: Unnecessary use of -X or --request, POST is already inferred.`, so we could remove the `-X POST` I think, (not just the `-X` like I did before 🤦, sorry), but I think I like being explicit in these examples. No strong feelings, though, so if we'd rather remove the `-X POST` entirely I'm okay with that, too

File: README.md (hunk 4 above)
Comment (mikelehen) [reply README.md]: Oh!  I see.  That worked, though TBH it's still a little confusing because it tells me it imported stuff but it still leaves me looking at the empty "New document" that I guess it created by default.  So I'd maybe put the instructions back how they were before but prefix with "Go to the main dashboard screen (e.g. by clicking "Insomnia" at the top of the screen)." or something like that.  Not a big deal though.  If I had played around with the tool a bit more I would have figured out.  It was just confusing / disorienting at first.

**Review by mikelehen — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (mikelehen) [reply README.md]: Oh, I see!  Yeah, no strong feelings.  I'm fine either way.

File: README.md (hunk 4 above)
Comment (smcclure17) [reply README.md]: Ah, interesting. Sounds good! I've reverted the docs and added the prefix 🙏
